funcx_endpoint/funcx_endpoint/endpoint/endpoint.py

```diff
@@ -139,15 +139,20 @@ def configure_endpoint(
         )
 
     @staticmethod
-    def check_endpoint_json(endpoint_json, endpoint_uuid):
-        if os.path.exists(endpoint_json):
-            with open(endpoint_json) as fp:
-                log.debug("Connection info loaded from prior registration record")
-                reg_info = json.load(fp)
-                endpoint_uuid = reg_info["endpoint_id"]
-        elif not endpoint_uuid:
-            endpoint_uuid = str(uuid.uuid4())
-        return endpoint_uuid
+    def get_endpoint_id(endpoint_dir: pathlib.Path) -> str | None:
+        info_file_path = endpoint_dir / "endpoint.json"
+        try:
+            return json.loads(info_file_path.read_bytes())["endpoint_id"]
+        except FileNotFoundError:
+            pass
+        return None
+
+    @staticmethod
+    def check_endpoint_json(endpoint_dir: pathlib.Path, endpoint_uuid: str) -> str:
+        ep_id = Endpoint.get_endpoint_id(endpoint_dir)
+        if not ep_id:
+            ep_id = endpoint_uuid or str(uuid.uuid4())
+        return ep_id
```

Comment:
Should we rename this? It's now deferring the actual json checks to `get_endpoint_id` - it's more of a "get endpoint id if it already exists, create if it doesn't" than a "check the endpoint json for an id or create one"

Reply:
Yes might as well rename to make it more clear.  Some version of 'get_or_generate_endpoint_id' though that skips the "use this uuid if there isn't an existing one"

Reply:
Yeah; a good time to rename for clarity.  Hang tight ...

Reply:
I don't really like the larger structure of how we've put this piece in, but I'm not going for a heavy refactor at this time.  Just little bits to make the future changes clearer.  In any event; please see if this works for y'all.

Reply:
>I don't really like the larger structure of how we've put this piece in

Agreed - realized how this function feels "off" as I was thinking through potential name suggestions